externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/GetFunctions.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine.flink.operation
+
+import scala.collection.JavaConverters._
+
+import org.apache.commons.lang3.StringUtils
+import org.apache.flink.table.api.{DataTypes, ResultKind, TableEnvironment}
+import org.apache.flink.table.catalog.Column
+import org.apache.flink.types.Row
+
+import org.apache.kyuubi.engine.flink.result.ResultSet
+import org.apache.kyuubi.engine.flink.util.StringUtils.filterPattern
+import org.apache.kyuubi.operation.OperationType
+import org.apache.kyuubi.operation.meta.ResultSetSchemaConstant._
+import org.apache.kyuubi.session.Session
+
+class GetFunctions(
+    session: Session,
+    catalogName: String,
+    schemaName: String,
+    functionName: String)
+  extends FlinkOperation(OperationType.GET_FUNCTIONS, session) {
+
+  override protected def runInternal(): Unit = {
+    try {
+      val schemaPattern = toJavaRegex(schemaName)
+      val functionPattern = toJavaRegex(functionName)
+      var rows: List[Row] = List()
+      val tableEnv: TableEnvironment = sessionContext.getExecutionContext.getTableEnvironment
+      filterPattern(
+        tableEnv.listFunctions().diff(tableEnv.listUserDefinedFunctions()),
+        functionPattern)
+        .foreach { f =>
+          rows = Row.of(null, null, f) +: rows
+        }
```

Comment:
We can leverage `map` `flatMap` to simplify the code.
```suggestion
    val globalFunctions = filterPattern(
        tableEnv.listFunctions().diff(tableEnv.listUserDefinedFunctions()),
        functionPattern)
        .map { f => Row.of(null, null, f) }
```